pkg/util/ecs/metadata/detection.go

```diff
@@ -137,3 +140,56 @@ func getAgentV4URLFromEnv() (string, error) {
 	}
 	return agentURL, nil
 }
+
+// IsMetadataV4Available checks if the v4 metadata endpoint is available.
+// It checks if metadata v4 url is available in the environment variable.
+// It should be available when the agent is deployed by ECS, as the ECS agent injects this environment variable.
+// Then it checks if agent is deployed directly on EC2 running in ECS.
+// It returns true if any container has the v4 endpoint env variable.
+func IsMetadataV4Available() (bool, error) {
+	// Agent is deployed by ECS
+	_, err := getAgentV4URLFromEnv()
+	if err == nil {
+		return true, nil
+	}
+
+	// If the agent is deployed on Fargate and v4 is not available, return false
+	if config.IsECSFargate() {
+		return false, errors.New("v4 endpoint not found in Fargate")
+	}
+
+	// Check if agent is deployed directly on EC2
+	du, err := docker.GetDockerUtil()
```

Comment:
Could you please change the name from `du` to something easier to follow? Maybe `dockerUtil`?

Reply:
[dafddab](https://github.com/DataDog/datadog-agent/pull/28007/commits/dafddab16dafd8e2ab5f97ef9d8daf2e7bb5f4e9)